src/workerd/jsg/resource.h

```diff
@@ -802,12 +804,43 @@ struct JsSetup {
 
   KJ_DISALLOW_COPY_AND_MOVE(JsSetup);
 
-  JsSetup(jsg::Lock& js): js(js) {}
+  JsSetup(jsg::Lock& js, v8::Local<v8::Context> context): js(js), context(context) {}
 
   inline void registerJsBundle(Bundle::Reader bundle) {
     ModuleRegistryImpl<TypeWrapper>::from(js)->addBuiltinBundle(bundle);
   }
 
+  template<const char* propertyName, const char* moduleName>
+  struct LazyJsInstancePropertyCallback {
+    static void callback(v8::Local<v8::Name> property,
+                         const v8::PropertyCallbackInfo<v8::Value>& info) {
+      static auto path = kj::Path::parse(moduleName);
+
+      auto& js = Lock::from(info.GetIsolate());
+      auto context = js.v8Context();
+      auto& moduleInfo = KJ_REQUIRE_NONNULL(
```

Comment:
I think you need to wrap the body of this function in `liftKj()`, otherwise any thrown exceptions will be fatal.

Reply:
hm... this actually breaks the functionality. Let me see why.

Reply:
nm, I was using it wrong.